### DISCUSSION TITLE
Got a error while import SWR

### DISCUSSION THREAD

**Discussion by marklee1972** [vercel/swr#1867] [Q&A] [answered]
I tried to use SWR in my next.js app ,but it's not work.
For excluding other variables, I install a pure page (index.js)from next.js example.
And it's fine to work,until I add a line on the head "import useSWR from 'swr'".

npm run dev console appeared :
_error - SyntaxError: The requested module 'react' does not provide an export named 'useRef'_

My browser appeared:

_Server Error
SyntaxError: The requested module 'react' does not provide an export named 'useRef'

This error happened while generating the page. Any console logs will be displayed in the terminal window.
Call Stack
<unknown>
file:///C:/Users/mark/Downloads/react/fo/nextjs-layout/node_modules/swr/dist/index.mjs (1)
ModuleJob._instantiate
internal/modules/esm/module_job.js (92:21)
async ModuleJob.run
internal/modules/esm/module_job.js (107:20)
async Loader.import
internal/modules/esm/loader.js (179:24)
async importModuleDynamicallyWrapper
internal/vm/module.js (422:15)_

I have no idea to debug,
Can someone give me any suggestion?

My package.json:
_{
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "next": "latest",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "swr": "^1.2.2"
  }
}_

**Comment by TLM-Dev** [comment, marked as answer]
I suspect it has something to do with the newer update of SWR.

I just had and solved this problem temporarily;
* deleting my node_modules folder.
* changing "swr": "^1.2.2" to "swr": "1.1.2" in package.json (the version of the example on their website, which is currently working).
* running "npm install" to reinstall the packages from the packages.json file.

I can't speak for what exactly the problem is on the 1.2.2 version or how to fix it, though. 
Hope this helps.

**Reply by marklee1972** [reply]
Great!!!
You got it !.
It's really the problem of swr version.
Should we report this issue/bug to Vercel?

**Reply by huozhi** [reply]
Sorry for the late visit, next.js should be able to resolve the proper assets of swr correctly now, I tired the latest next (12.2.4) at the moment it doesn't have issue with swr latest (1.3.0)

**Comment by iShubhamPrakash** [comment]
I'm having this issue in swr v2.2.5
<img width="1167" alt="Screenshot 2024-07-03 at 4 24 36 PM" src="https://github.com/vercel/swr/assets/28767301/731af333-3dd1-4a0a-9f3f-e1d1dd6dcfb5">

**Reply by SystemDisc** [reply]
Same issue here. I reverted to using 2.2.4 for now: `npm i swr@2.2.4`
